File: c++ DroneSim/eccezione.cpp
#include "eccezione.h"
#include <iostream>
using std::cout;

eccezione::eccezione(char c): quale(c){}

void eccezione::stampa() const {
    switch(quale){
    case('b'):
        cout<<"batteria esaurita\n";
        break;
    case('m'):
        cout<<"mossa non valida\n";
        break;
    case('s'):
        cout<<"accensione/spegnimento già effettuato\n";
        break;
    case('v'):
        cout<<"Valori non validi\n";
        break;
    default:
        cout<<"Eccezione generica\n";
        break;
    }
}

File: java DroneSim/Eccezione.java
public class Eccezione extends Exception{

    private char quale;

    public Eccezione(char c){
    	quale = c;
    }

    public void stampa(){
    	switch(quale){
    		case('b'):
		        System.out.println("Batteria esaurita");
		        break;
		    
		    case('m'):
		        System.out.println("Mossa non valida");
		        break;
		    case ('s'):
		    	System.out.println("Accensione/spegnimento già eseguito");
		    	break;
		    case('v'):
		        System.out.println("Valori non validi");
		        break;
		    default:
		        System.out.println("Eccezione generica");
		        break;
	    }
    }

}